django/contrib/sitemaps/__init__.py

```diff
@@ -202,23 +209,25 @@ def _urls(self, page, protocol, domain):
 
             if self.i18n and self.alternates:
                 for lang_code in self._languages():
-                    loc = f"{protocol}://{domain}{self._location(item, lang_code)}"
-                    url_info["alternates"].append(
-                        {
-                            "location": loc,
-                            "lang_code": lang_code,
-                        }
-                    )
+                    if self.enable_item_for_lang(item[0], lang_code):
```

Comment:
What do you think about adding a method with languages for the item instead? E.g.
```python
def get_languages_for_item(self, item):
    return self._languages()

def _urls(self, page, protocol, domain):
    ...
    if self.i18n and self.alternates:
        for lang_code in self.get_languages_for_item(item):
```

Reply:
Love it !